logstash-core/src/main/java/org/logstash/execution/LogstashPipelineExt.java

```diff
@@ -159,7 +219,93 @@ public IRubyObject closeDlqWriter(final ThreadContext context) {
         return context.nil;
     }
 
+    @JRubyMethod
+    public PipelineReporterExt reporter() {
+        return reporter;
+    }
+
+    @JRubyMethod(name = "collect_dlq_stats")
+    public IRubyObject collectDlqStats(final ThreadContext context) {
+        if (dlqEnabled(context).isTrue()) {
+            getDlqMetric(context).gauge(
+                context, DLQ_SIZE_KEY,
+                dlqWriter(context).callMethod(context, "get_current_queue_size")
+            );
+        }
+        return context.nil;
+    }
+
+    @JRubyMethod(name = "system?")
+    public IRubyObject isSystem(final ThreadContext context) {
+        return getSetting(context, "pipeline.system");
+    }
+
+    @JRubyMethod(name = "configured_as_reloadable?")
+    public IRubyObject isConfiguredReloadable(final ThreadContext context) {
+        return getSetting(context, "pipeline.reloadable");
+    }
+
+    @JRubyMethod(name = "collect_stats")
+    public IRubyObject collectStats(final ThreadContext context) throws IOException {
+        final AbstractNamespacedMetricExt pipelineMetric = metric.namespace(
+            context,
+            RubyArray.newArray(
+                context.runtime,
+                Arrays.asList(STATS_KEY, PIPELINES_KEY, pipelineId.asString().intern(), QUEUE_KEY)
+            )
+        );
+        pipelineMetric.gauge(context, TYPE_KEY, getSetting(context, "queue.type"));
+        if (queue instanceof JRubyWrappedAckedQueueExt) {
```

Comment:
It would be nice if collection of queue-specific stats could be delegated to the queue class itself. That's beyond the scope of this PR, though.